networkx/algorithms/community/modularity_max.py

```diff
@@ -4,22 +4,24 @@
 
 from networkx.utils.mapped_queue import MappedQueue
 
+import warnings
+
 __all__ = [
     "greedy_modularity_communities",
     "naive_greedy_modularity_communities",
     "_naive_greedy_modularity_communities",
 ]
 
 
-def greedy_modularity_communities(G, weight=None, resolution=1):
+def greedy_modularity_communities(G, weight=None, resolution=1, n_communities=1):
     r"""Find communities in G using greedy modularity maximization.
 
     This function uses Clauset-Newman-Moore greedy modularity maximization [2]_.
     This method currently supports the Graph class.
 
     Greedy modularity maximization begins with each node in its own community
     and joins the pair of communities that most increases modularity until no
-    such pair exists.
+    such pair exists or until desired number of communities is reached.
```

Comment:
```suggestion
    such pair exists or until `n_communities` number of communities is reached.
```
Helps to give users a feel for what this parameter does in the summary.